tests/test_scripts.py

```diff
@@ -82,3 +84,123 @@ def test_proto_conversion(tmpdir):
     assert p.returncode == 0
 
     assert p.stdout.startswith(b'{\n  "meta": ') or p.stdout.startswith(b'{\r\n  "meta": ')
+
+
+def run_detect_duplicate_features(rule_dir, rule_path):
+    args = [rule_dir, rule_path]
+    script_path = get_script_path("detect_duplicate_features.py")
+    args = [sys.executable] + [script_path] + args
+    print(f"running: '{args}'")
+    return subprocess.run(args)
+
+
+def test_detect_duplicate_features(tmpdir):
+    TEST_RULE_0 = textwrap.dedent(
+        """
+        rule:
+            meta:
+                name: Test Rule 0
+                scope: function
+            features:
+              - and:
+                - number: 2
+                - or:
+                  - mnemonic: shr
+                  - api: connect
+        """
+    )
+
+    TEST_RULESET = {
+        "rule_1": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 1
+                        scope: function
+                    features:
+                      - or:
+                        - string: "sites.ini"
+                        - number: 0xEDB88320
+                        - and:
+                          - or:
+                            - arch: i386
+                            - number: 4
+                            - not:
+                              - count(mnemonic(xor)): 5
+                          - not:
+                            - os: linux
+            """
+        ),
+        "rule_2": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 2
+                        scope: function
+                    features:
+                      - and:
+                        - string: "sites.ini"
+                        - arch: i386
+                        - basic block:
+                          - and:
+                            - api: setsockopt
+                            - count(mnemonic(mov)): 3
+            """
+        ),
+        "rule_3": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 3
+                        scope: function
+                    features:
+                      - or:
+                        - not:
+                          - os: linux
+                        - basic block:
+                          - and:
+                            - api: bind
+                            - count(mnemonic(mov)): 3
+            """
+        ),
+        "rule_4": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 4
+                        scope: function
+                    features:
+                      - not:
+                        - string: "expa"
+            """
+        ),
+    }
+
+    """
+        The rule_overlaps list represents the number of overlaps between each rule in the RULESET.
```

Comment:
can you name/comment the rules to better indicate this or indicate easily so that we understand these tests in a few weeks :)

Reply:
Do you mean to add some comments on what features are matching between rules.